### DISCUSSION TITLE
Update vs .show

### DISCUSSION THREAD

**Discussion by hafezrouzati** [emilk/egui#2823] [General]
In the Widget Gallery example, the code calls the .ui function from inside the show() method whereas in the eframe_template, there is an update method where all the widgets are to be added. 

Is there a difference in these two approaches?  For example, is the show() method also called constantly to re-render the UI?

I'm currently using the eframe_template to build my application.

Thanks

**Comment by stonerfish** [comment]
eframe will call update() for you each frame.  Your update() can call show().
In the egui_demo_app, the only function called each frame from eframe is the wrap_app::update()
The other update() methods are all called from the wrap_app::update().  The show() methods are for code organization.
